src/test/java/picard/vcf/LiftoverVcfTest.java

```diff
@@ -159,12 +138,345 @@ public void testWriteOriginalPosition(boolean shouldWriteOriginalPosition) {
+        }
+
+        //TCTC/TC-> ACT/A in CT repeat region
+        for (start = 21; start <= 29; start += 2) {
+            builder.start(start).stop(start + 3);
+            builder.alleles(CollectionUtil.makeList(
+                    RefTCTC, TC));
+
+            tests.add(new Object[]{builder.make(), reference, result_builder.make()});
+        }
+
+        // for ease of reading, here's the reference sequence
+        // "CAAAAAAAAAACGTACGTACTCTCTCTCTACGT"
+        //  123456789 123456789 123456789 123
+
+        result_builder.alleles("AACGT", "A").start(10).stop(14);
+        for (start = 10; start < 17; start++) {
+            for (stop = start + 4; stop < 20; stop++) {
+                builder.alleles(
+                        // -1 here due to reference string being 0-based.
+                        reference.getBaseString().substring(start - 1, stop + 1 - 1),
+                        reference.getBaseString().substring(start - 1, stop - 3 - 1)).start(start).stop(stop);
+                tests.add(new Object[]{builder.make(), reference, result_builder.make()});
+            }
+        }
+
+        // test vc with genotypes:
+
+        result_builder.start(10).stop(14).alleles("AACGT", "A", "AACG", "AACGTACGT");
+        builder.start(9).stop(18).alleles("AAACGTACGT", "AAACGT", "AAACGACGT", "AAACGTACGTACGT");
+        Collection<Genotype> genotypes = new ArrayList<>();
+        Collection<Genotype> results_genotypes = new ArrayList<>();
+        GenotypeBuilder gBuilder = new GenotypeBuilder();
+
+        for (int sample = 1; sample < 4; sample++) {
+            gBuilder.name("sample" + sample)
+                    .alleles(CollectionUtil.makeList(builder.getAlleles().get(0), builder.getAlleles().get(sample)));
+            genotypes.add(gBuilder.make());
+            gBuilder.alleles(CollectionUtil.makeList(result_builder.getAlleles().get(0), result_builder.getAlleles().get(sample)));
+            results_genotypes.add(gBuilder.make());
+        }
+        gBuilder.name("sample_non_ref_het")
+                .alleles(CollectionUtil.makeList(builder.getAlleles().get(1), builder.getAlleles().get(2)));
+        genotypes.add(gBuilder.make());
+        gBuilder.alleles(CollectionUtil.makeList(result_builder.getAlleles().get(1), result_builder.getAlleles().get(2)));
+        results_genotypes.add(gBuilder.make());
+
+        builder.genotypes(genotypes);
+        result_builder.genotypes(results_genotypes);
+
+        tests.add(new Object[]{builder.make(), reference, result_builder.make()});
+
+        return tests.iterator();
+    }
+
+    @Test(dataProvider = "leftAlignAllelesData")
+    public void testLeftAlignVariants(final VariantContext source, final ReferenceSequence reference, final VariantContext result) {
+
+        assertVcAreEqual(leftAlignVariant(source, reference), result);
+    }
+
+    private void assertVcAreEqual(final VariantContext actual, final VariantContext expected) {
```

Comment:
why can't you just use Assert.equals(actual, expected)?  That should work.  These tests are already in the VC class I believe.